icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameHandler.java

```diff
@@ -609,6 +609,102 @@ private static String getDeriveCompoundRule(ULocale locale, String feature, Stri
     /// END DATA LOADING ///
     ////////////////////////
 
+    private static String calculateGenderForUnit(ULocale locale, MeasureUnit unit) {
+        // The "forArbitraryUnit" code path calculates gender for every component of
+        // a compound unit, even if unneeded. This algorithm does a lazier evaluation.
+
+        MeasureUnitImpl mui = unit.getCopyOfMeasureUnitImpl();
+        ArrayList<SingleUnitImpl> singleUnits = mui.getSingleUnits();
+        int singleUnitIndex = 0;
+        if (mui.getComplexity() == MeasureUnit.Complexity.COMPOUND) {
+            int startSlice = 0;
+            // inclusive
+            int endSlice = singleUnits.size() - 1;
+            assert endSlice > 0 : "COMPOUND units have more than one single unit";
```

Comment:
this assert could be moved to the MeasureUnit to ensure the compound units have more than one unit. I think it is too much to have this assert here

Reply:
The point of this assert is to locally document the invariants. I feel as far as this PR is concerned, MeasureUnit does what it does, we trust it or we don't trust it - so I could delete the assert, or keep the assert, no need to move it to MeasureUnit.

What this bit of documentation does, is alleviate the reader from wondering whether the value produced by the previous line, `int endSlice = singleUnits.size() - 1;`, is valid. What if size() is zero, and endSlice is thus -1? And then I call `singleUnits.get(endSlice)` in the next line? What does Java do when you call get(index) with index being out of bounds for an array? I don't know - ideally I would know, but this assert makes it clear that for the purpose of this code, I don't need to know.

Do you still think this assert is too much? Did you wonder about the value of endSlice when you reviewed this code?